azure/services/virtualmachines/virtualmachines.go

```diff
@@ -436,8 +440,24 @@ func (s *Service) generateStorageProfile(ctx context.Context, vmSpec azure.VMSpe
 
 			// check the support for ultra disks based on location and vm size
 			location := s.Scope.Location()
-			if disk.ManagedDisk.StorageAccountType == UltraSSDStorageAccountType && !sku.HasLocationCapability(resourceskus.UltraSSDAvailable, location, vmSpec.Zone) {
-				return nil, azure.WithTerminalError(fmt.Errorf("vm size %s does not support ultra disks in location %s. select a different vm size or disable ultra disks", vmSpec.Size, location))
+			if disk.ManagedDisk.StorageAccountType == UltraSSDStorageAccountType {
+				if !sku.HasLocationCapability(resourceskus.UltraSSDAvailable, location, vmSpec.Zone) {
+					return nil, azure.WithTerminalError(fmt.Errorf("vm size %s does not support ultra disks in location %s. select a different vm size or disable ultra disks", vmSpec.Size, location))
+				}
+				// Configure max IOPS and throughput as per this reference:
+				// https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types#disk-size
+				iops := int(disk.DiskSizeGB) * UltraSSDDiskSizeIOPSMultiplier
```

Comment:
This configuration appears to be read-only in fact, if you read the SDK comments:

https://github.com/Azure/azure-sdk-for-go/blob/main/services/compute/mgmt/2020-06-30/compute/models.go#L1147

I did some manual cluster builds and observed the IOPS and throughput config for the etcd data disk was still the default.

I was able to manually change the value in the portal:

<img width="842" alt="image" src="https://user-images.githubusercontent.com/1494193/130302330-28ae01d1-81fd-41d3-b169-7bd2266ddc2a.png">

Perhaps there's an API to do that async after the VM and data disk attachment has been made?

Otherwise as far as I can tell we get a disk with UltraSSD capabilities, but tuned for very low throughput:

<img width="860" alt="image" src="https://user-images.githubusercontent.com/1494193/130302379-2674f121-6f69-405d-a7c4-5d93cd0d2ed9.png">

cc @devigned